sparkmagic/sparkmagic/controllerwidget/manageendpointwidget.py

```diff
@@ -122,3 +127,12 @@ def get_info_endpoint_widget(self, endpoint, url):
             text = "No sessions on this endpoint."
 
         return self.ipywidget_factory.get_html(text, width=width)
+
+    def initialize_kerberos(self, endpoint):
+        if endpoint.auth_type == constants.AUTH_KERBEROS and self.kerberos_info.get(endpoint.username) is None:
+            # Stop flag can be used to stop renewing kerberos ticket for the user
+            stop_flag = Event()
+            KerberosThread(stop_flag, endpoint).start()
+            self.kerberos_info[endpoint.username] = stop_flag
```

Comment:
How can the user use the `stop_flag`? This sounds like something we could automate for the user as soon as the kernel dies or the endpoint is removed?